### PR TITLE
MP-1136 Upgrade numpy

### PR DESCRIPTION
**Pull request by TimurDaukaevMoonvision**
## Description
I want to upgrade `scipy` and `scikit-learn` in other repos, so that I could run tests for Python 3.11, and for this I want to make sure, that all environments have the same `numpy` version. So I'm copying the newer version from https://github.com/MoonVision/moonbox-docker/pull/12.

This should also eventually fix the warning about mismatching numpy versions between client, scheduler and workers.

In addition I am fixing broken `pytorch` builds: our current `pytorch` version is not compatible with the latest `setuptools`, but it does not set an upper boundary.

## How to test
Builds are successful.

### CODE REVIEW

**Review by ahirner — APPROVED** [review]
Thanks. It will be interesting to see what computer vision (low level flakeyness or deprecations) it hits in vid-vision.
